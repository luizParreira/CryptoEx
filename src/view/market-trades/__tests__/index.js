import React from 'react';
import renderer from 'react-test-renderer';
import {MarketTrades} from '..';

describe('MarketTrades', () => {
  const navigation = {
    navigate: jest.fn()
  };

  const buyOrder = {id: 546, type: 'buy', quantity: 9, price: 228, formattedPrice: '$ 228.0'};
  const sellOrder = {id: 547, type: 'sell', quantity: 5, price: 129, formattedPrice: '$ 129.0'};
  const trades = [
    {
      price: 10,
      quantity: 2,
      buyOrder,
      sellOrder,
      time: 12233444567,
      formattedPrice: '$ 10.0',
      formattedTime: '2:22:10 pm'
    },
    {
      price: 10,
      quantity: 2,
      buyOrder,
      sellOrder,
      time: 12233444567,
      formattedPrice: '$ 10.0',
      formattedTime: '2:22:10 pm'
    }
  ];

  const props = {trades, navigation};

  test('renders successfully with trades', () => {
    const expectedComponent = renderer.create(<MarketTrades {...props} />);

    expect(expectedComponent).toMatchSnapshot();
  });

  test('renders successfully with empty trades', () => {
    const expectedComponent = renderer.create(<MarketTrades {...{...props, trades: []}} />);

    expect(expectedComponent).toMatchSnapshot();
  });
});
