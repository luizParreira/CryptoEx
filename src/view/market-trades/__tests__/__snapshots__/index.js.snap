// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketTrades renders successfully with empty trades 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#c6c4c4",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            textAlign="left"
          >
            Time
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            textAlign="center"
          >
            Price
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "right",
                },
              ]
            }
            textAlign="right"
          >
            Amount
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`MarketTrades renders successfully with trades 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#c6c4c4",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            textAlign="left"
          >
            Time
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            textAlign="center"
          >
            Price
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c6c4c4",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "right",
                },
              ]
            }
            textAlign="right"
          >
            Amount
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#c6c4c4",
                "borderBottomWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "left",
                  },
                ]
              }
              textAlign="left"
            >
              2:22:10 pm
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
              textAlign="center"
            >
              $ 10.0
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "right",
                  },
                ]
              }
              textAlign="right"
            >
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#c6c4c4",
                "borderBottomWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "left",
                  },
                ]
              }
              textAlign="left"
            >
              2:22:10 pm
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
              textAlign="center"
            >
              $ 10.0
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "textAlign": "right",
                  },
                ]
              }
              textAlign="right"
            >
              2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
